nullaway/src/main/java/com/uber/nullaway/handlers/AssertionHandler.java

```diff
@@ -0,0 +1,128 @@
+      AccessPathNullnessPropagation.Updates bothUpdates) {
+    Symbol.MethodSymbol callee = ASTHelpers.getSymbol(node.getTree());
+    if (callee == null) {
+      return NullnessHint.UNKNOWN;
+    }
+
+    if (!areMethodNamesInitialized()) {
+      initializeMethodNames(callee);
+    }
+
+    /**
+     * Look for statements of the form: assertThat(A).isNotNull()
+     *
+     * <p>A will not be NULL after this statement.
+     */
+    if (isMethodIsNotNull(callee)) {
+      Node receiver = node.getTarget().getReceiver();
+      if (receiver instanceof MethodInvocationNode) {
+        MethodInvocationNode receiver_method = (MethodInvocationNode) receiver;
+        Symbol.MethodSymbol receiver_symbol = ASTHelpers.getSymbol(receiver_method.getTree());
+        if (isMethodAssertThat(receiver_symbol)) {
+          Node arg = receiver_method.getArgument(0);
+          AccessPath ap = AccessPath.getAccessPathForNodeNoMapGet(arg);
+          if (ap != null) {
+            bothUpdates.set(ap, NONNULL);
+          }
+        }
+      }
+    }
+    return NullnessHint.UNKNOWN;
+  }
+
+  private boolean isMethodIsNotNull(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, isNotNull, isNotNullOwner);
+  }
+
+  private boolean isMethodAssertThat(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, assertThat, assertThatOwner);
+  }
+
+  private boolean matchesMethod(
+      Symbol.MethodSymbol methodSymbol, Name toMatchMethodName, Name toMatchOwnerName) {
+    return methodSymbol.name.equals(toMatchMethodName)
+        && methodSymbol.owner.getQualifiedName().equals(toMatchOwnerName);
+  }
+
+  private boolean areMethodNamesInitialized() {
+    return isNotNull != null;
+  }
+
+  private synchronized void initializeMethodNames(Symbol.MethodSymbol methodSymbol) {
+    isNotNull = methodSymbol.name.table.fromString(IS_NOT_NULL_METHOD);
+    isNotNullOwner = methodSymbol.name.table.fromString(IS_NOT_NULL_OWNER);
+    assertThat = methodSymbol.name.table.fromString(ASSERT_THAT_METHOD);
+    assertThatOwner = methodSymbol.name.table.fromString(ASSERT_THAT_OWNER);
+  }
+
+  /**
+   * Strings corresponding to the names of the methods (and their owners) used to identify
+   * assertions in this handler.
+   */
+  private static final String IS_NOT_NULL_METHOD = "isNotNull";
+
+  private static final String IS_NOT_NULL_OWNER = "com.google.common.truth.Subject";
+  private static final String ASSERT_THAT_METHOD = "assertThat";
+  private static final String ASSERT_THAT_OWNER = "com.google.common.truth.Truth";
+
+  /**
+   * Names of the methods (and their owners) used to identify assertions in this handler. Name used
+   * here refers to com.sun.tools.javac.util.Name. Comparing methods using Names is faster than
+   * comparing using strings.
+   */
+  private Name isNotNull;
+
+  private Name isNotNullOwner;
+  private Name assertThat;
+  private Name assertThatOwner;
```

Comment:
Let's move the fields up. 

Both NullAway and internal Java code tends to follow the ordering: Fields (static and instance) -> Constructors -> Methods

Reply:
Agreed

Reply:
Done.